master/pkg/searcher/searcher.go

```diff
@@ -109,6 +109,15 @@ func (s *Searcher) TrialExitedEarly(
 		return nil, errors.Wrapf(err, "error relaying trial exited early to trial %d", requestID)
 	}
 	s.Exits[requestID] = true
+
+	_, isCustomSMethod := s.method.(CustomSearchMethod)
+	// for non custom-search methods you can make the assumption
+	// that trials will be created immediately.
+	if s.TrialsRequested == len(s.TrialsClosed) && !isCustomSMethod {
+		shutdown := Shutdown{Failure: len(s.Failures) >= s.TrialsRequested}
+		operations = append(operations, shutdown)
+	}
+
 	s.Record(operations)
```

Comment:
In the [fix made by Danny](https://github.com/determined-ai/determined/commit/2c575fd946a38f5e6e75eb2657b6d81b47d5b068) `s.Record(operations)` is invoked before checking for shutdown to ensure all operations are recorded. I think we should follow the same logic here.

Reply:
Yep, both places should have the logic from that fix.